streams/src/main/java/org/apache/kafka/streams/kstream/internals/suppress/TimeKey.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream.internals.suppress;
+
+import java.util.Objects;
+
+class TimeKey<K> implements Comparable<TimeKey<K>> {
+    private final long time;
+    private final K key;
+
+    TimeKey(final long time, final K key) {
+        this.time = time;
+        this.key = key;
+    }
+
+    K key() {
+        return key;
+    }
+
+    long time() {
+        return time;
+    }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        final TimeKey<?> timeKey = (TimeKey<?>) o;
+        return time == timeKey.time &&
+            Objects.equals(key, timeKey.key);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(time, key);
+    }
+
+    @Override
+    public int compareTo(final TimeKey<K> o) {
+        // ordering of keys within a time uses hashCode.
+        final int timeComparison = Long.compare(time, o.time);
+        return timeComparison == 0 ? Integer.compare(key.hashCode(), o.key.hashCode()) : timeComparison;
+    }
+
+    @Override
+    public String toString() {
+        return "TimeKey{time=" + time + ", key=" + key + '}';
```

Comment:
`key` is always `Bytes()`, thus, this output is not very useful. Can we can hold on the deserialized for human readable output here?

Reply:
I didn't notice that. That is handy.

Reply:
Similar argument as for `byte[]` value: Of course, here we still get the `time` information, but the `Bytes` `key` is useless.